src/keys/keypair.rs

```diff
@@ -0,0 +1,241 @@
+// Copyright 2019 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under the MIT license <LICENSE-MIT
+// https://opensource.org/licenses/MIT> or the Modified BSD license <LICENSE-BSD
+// https://opensource.org/licenses/BSD-3-Clause>, at your option. This file may not be copied,
+// modified, or distributed except according to those terms. Please review the Licences for the
+// specific language governing permissions and limitations relating to use of the SAFE Network
+// Software.
+
+//! Module providing keys, keypairs, and signatures.
+//!
+//! The easiest way to get a `PublicKey` is to create a random `Keypair` first through one of the
+//! `new` functions. A `PublicKey` can't be generated by itself; it must always be derived from a
+//! secret key.
+
+use crate::{Error, Result};
+use crate::{PublicKey, SecretKey, Signature, SignatureShare};
+
+use ed25519_dalek::Signer;
+use rand::{CryptoRng, Rng};
+use serde::{Deserialize, Serialize};
+use std::fmt::{self, Debug, Formatter};
+use threshold_crypto::{self, serde_impl::SerdeSecret};
+use unwrap::unwrap;
+
+/// Wrapper for different keypair types.
+#[derive(Serialize, Deserialize)]
+pub enum Keypair {
```

Comment:
Should this one be `Keypairs` plural?

Reply:
Not yet. It's just a wrapper around the various keypair types.

This isn't renamed, so I'd ideally not want to touch this here if we're going to refactor this all shortly anywho